encoding/kzg/prover/proof_backend.go

```diff
@@ -0,0 +1,18 @@
+package prover
+
+import (
+	"github.com/consensys/gnark-crypto/ecc/bn254"
+	"github.com/consensys/gnark-crypto/ecc/bn254/fr"
+)
+
+// Proof device represents a backend capable of computing KZG multiproofs.
+type KzgMultiProofsBackend interface {
+	ComputeMultiFrameProof(blobFr []fr.Element, numChunks, chunkLen, numWorker uint64) ([]bn254.G1Affine, error)
+}
+
+// CommitmentDevice represents a backend capable of computing various KZG commitments.
+type KzgCommitmentsBackend interface {
```

Comment:
Moved commitments to its own backend in case we would like to accelerate them in the future.